pokemon-service/src/test/java/cz/muni/fi/pa165/pokemon/league/participation/manager/service/PokemonServiceTest.java

```diff
@@ -0,0 +1,252 @@
+                .build();
+
+        trainerAsh = new TrainerBuilder()
+                .id(40L)
+                .born(LocalDate.of(1997, 7, 11))
+                .isAdmin(false)
+                .name("Ash")
+                .passwordHash("asdjahdkjfhs12312")
+                .surname("Ketchup")
+                .userName("Ash97")
+                .build();
+
+        trainerBrock = new TrainerBuilder()
+                .id(41L)
+                .born(LocalDate.of(1996, 8, 12))
+                .isAdmin(false)
+                .name("Brock")
+                .passwordHash("adj12jndakj3")
+                .surname("Noeye")
+                .userName("Brocky")
+                .build();
+
+        pokemonPikachu = new PokemonBuilder()
+                .id(100L)
+                .level(40)
+                .nickname("Pikachuchu")
+                .trainer(trainerAsh)
+                .pokemonSpecies(pokemonSpeciesPikachu)
+                .dateTimeOfCapture(LocalDateTime.of(2018, 10, 11, 5, 30))
+                .build();
+
+        pokemonCharizard = new PokemonBuilder()
+                .id(101L)
+                .level(95)
+                .nickname("Charry")
+                .trainer(trainerAsh)
+                .pokemonSpecies(pokemonSpeciesCharizard)
+                .dateTimeOfCapture(LocalDateTime.of(2018, 2, 2, 14, 23))
+                .build();
+
+        exceptionalPokemon = new PokemonBuilder()
+                .id(102L)
+                .level(1000)
+                .nickname("Hacked Pokemon")
+                .trainer(trainerBrock)
+                .pokemonSpecies(pokemonSpeciesPichu)
+                .dateTimeOfCapture(LocalDateTime.of(2018, 12, 1, 22, 48))
+                .build();
+
+        when(pokemonDAO.findPokemonById(pokemonPikachu.getId())).thenReturn(pokemonPikachu);
+        when(pokemonDAO.findPokemonById(pokemonCharizard.getId())).thenReturn(pokemonCharizard);
+        when(pokemonDAO.findPokemonById(exceptionalPokemon.getId())).thenThrow(PE);
+        when(pokemonDAO.getPokemonOfTrainer(trainerAsh)).thenReturn(Stream.of(pokemonPikachu, pokemonCharizard).collect(Collectors.toList()));
+        when(pokemonDAO.getAllPokemonOfSpecies(pokemonSpeciesPikachu)).thenReturn(Stream.of(pokemonPikachu).collect(Collectors.toList()));
+        when(pokemonDAO.getAllPokemonOfSpecies(pokemonSpeciesCharizard)).thenReturn(Stream.of(pokemonCharizard).collect(Collectors.toList()));
+        when(pokemonDAO.getAllPokemonOfSpecies(pokemonSpeciesCharmander)).thenReturn(new ArrayList<>());
+        when(pokemonDAO.getAllPokemonOfSpecies(pokemonSpeciesCharmeleon)).thenReturn(new ArrayList<>());
+        when(pokemonDAO.getPokemonOfTrainer(trainerBrock)).thenThrow(PE);
+        doThrow(PE).when(pokemonDAO).updatePokemon(exceptionalPokemon);
+        doThrow(PE).when(pokemonDAO).deletePokemon(exceptionalPokemon);
+    }
+
+    @Test
+    public void testCreatePokemon() {
+        doThrow(PE).when(pokemonDAO).createPokemon(exceptionalPokemon);
```

Comment:
I think it's safe to put this in set up. The TrainerService tests put it here, because elsewhere they need a valid password hash even for the exceptionalTrainer (so they need a createTrainer call that will succeed).